scripts/build/Platform/Linux/deploy_cdk_applications.sh

```diff
@@ -62,8 +62,9 @@ then
 fi
 
 echo [cdk_installation] Install the current version of nodejs
-curl -fsSL https://deb.nodesource.com/setup_current.x | sudo -E bash -
-sudo apt-get install -y nodejs
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/$NVM_VERSION/install.sh | bash
+source ~/.bashrc
+nvm install node
 
 echo [cdk_installation] Install the latest version of CDK
 if ! sudo npm uninstall -g aws-cdk;
```

Comment:
Not related to this PR: if sudo is not recommended for security concern, should we use python to install aws cdk as well?

Reply:
Nothing wrong with continuing to use shell if it makes sense. 

btw, to expand on my last comment: The issue isn't that those instructions from nodesource run sudo (though it should be avoided as much as possible), it's that it's *piping* a run script directly from source with sudo. This can still happen if you did this from Python, but you'd have more tools to verify that the install script is valid.

Aside from the obvious issue (a MitM or supply chain attack running without restriction), If that script runs something like `rm -rf /directory` and it fails to download completely, there is potential for that pipe command to execute `rm -rf /` without restriction.